packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -0,0 +1,63 @@
+import fetch, { RequestInit } from 'node-fetch';
+
+type GenericDict = { [ key: string ]: string | number };
+type RequestMethods = 'post' | 'get';
+type RequestParams = {
+	method: 'post' | 'get';
+	headers: GenericDict;
+	body?: GenericDict;
+};
```

Comment:
Typing is not my strength and I struggled somewhat to come up with these type definitions.

I read up on Generics in TypeScript again but I'm not certain if this definition of `GenericDict` is in fact a generic type.

Reply:
What you have works! But I think it differs a little bit from TypeScript generics, which is more about passing in different types almost as arguments to a function. The docs go into it a little bit more [here](https://www.typescriptlang.org/docs/handbook/2/generics.html).

The idea would be then that you could define whatever interfaces you wanted for the headers and body, and pass them into the `sendRequest` function:

```typescript
private async sendRequest<Header, Body>( url: string, params: RequestParams<Header, Body> )
```